kong/cache.lua

```diff
@@ -31,6 +31,18 @@ local function log(lvl, ...)
   return ngx_log(lvl, "[DB cache] ", ...)
 end
 
+-- Temporary fix to convert soft callback errors into hard ones.
+-- FIXME: use upstream mlcache lib instead of local copy
+local soft_to_hard = function(cb)
+  return function(...)
```

Comment:
perf: it would probably be worth it to maintain a cache of such closures to lower the GC overhead and ensure we JIT compile in such hot code paths served by the caching library (performance being the whole point here). This change as it is would be quite hurtful to performance.